### DISCUSSION TITLE
Best practices for running ETS alongside hkknx

### DISCUSSION THREAD

**Discussion by dotgrid** [brutella/hkknx-public#151] [Q&A]
Are there any best practices to follow when working with ETS on the local KNX network, alongside an active hkknx server?

- Is it best to _shut down hkknx_ whilst ETS "downloads" settings/applications to KNX devices?
- After uploading an updated .knxproj file in hkknx, should previously uploaded KNX projects be deleted? (Especially when group addresses have been changed/removed?)
- Are there recommended config settings for the _KNX IP Router_ that hkknx connects to? Required features or things to switch off, for best performance?
- Anything else that is good to know 🙂 

Thanks!

**Comment by brutella** [comment]
> * Is it best to _shut down hkknx_ whilst ETS "downloads" settings/applications to KNX devices?

I don't think this is necessary.

> * After uploading an updated .knxproj file in hkknx, should previously uploaded KNX projects be deleted? (Especially when group addresses have been changed/removed?)

Yes, you can delete previously imported KNX project files and import the new one.
Deleting the project file doesn't change the HomeKit accessories.

> * Are there recommended config settings for the _KNX IP Router_ that hkknx connects to? Required features or things to switch off, for best performance?

I don't know of any.

> * Anything else that is good to know 🙂

- Run hkknx on a dedicated computer (eg Raspberry Pi) and connect it via Ethernet to your local network. Disable the Wifi interface.
Do not run other HomeKit (eg Homebridge) or KNX services (eg knxd) on that computer.

- Install [Configurator.app](https://apps.apple.com/app/id1525371546) to synchronise rooms and other stuff with HomeKit. This is currently only documented in the German version of the Online Handbook. → https://hochgatterer.me/de/hkknx/docs/accessories/#konfigurator-app
